pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -497,6 +500,40 @@ def _set_kargs(self):
 
         safe_exec_with_redirect("ostree", set_kargs_args, root=self._sysroot)
 
+        if arch.is_s390():
+            # Deployment was done. Enable ostree's zipl support; this is how things are currently done in e.g.
+            # https://github.com/coreos/coreos-assembler/blob/7d6fa376fc9f73625487adbb9386785bb09f1bb2/src/osbuild-manifests/coreos.osbuild.s390x.mpp.yaml#L261
+            safe_exec_with_redirect(
+                "ostree",
+                ["config",
+                 "--repo=" + self._sysroot + "/ostree/repo",
+                 "set", "sysroot.bootloader", "zipl"]
+            )
+
+            for bls_path in sorted(glob.glob(self._sysroot + "/boot/loader/entries/*.conf")):
+                log.info("found %s", bls_path)
+                with open(bls_path, "r") as bls:
+                    for line in bls.readlines():
+                        if line.startswith("options "):
+                            cmdline = line.split()[1]
+                        if line.startswith("linux"):
+                            kernel = self._sysroot + "/boot" + line.split()[1]
+                        if line.startswith("initrd"):
+                            initrd = self._sysroot + "/boot" + line.split()[1]
+                break
+
```

Comment:
I wonder about the outside `for` cycle here. What is the benefit of having the cycle with unconditional `break` at the end. If I'm reading it correctly the same job could be handled with and `if`?